syntax="proto3";

import "ECS/proto/components/transform_component.proto";
import "ECS/proto/components/visual_component.proto";
import "ECS/proto/components/input_component.proto";
import "ECS/proto/components/health_component.proto";
import "ECS/proto/components/camera_component.proto";
import "ECS/proto/components/terrain_component.proto";
import "ECS/proto/components/light_component.proto";
import "ECS/proto/components/script_component.proto";

package astre.ecs;

message EntityDefinition
{
    uint64 id = 1;
    string name = 2;

    optional TransformComponent transform = 3;
    optional HealthComponent health = 4;
    optional VisualComponent visual = 5;
    optional InputComponent input = 6;
    optional CameraComponent camera = 7;
    optional TerrainComponent terrain = 8;
    optional LightComponent light = 9;
    optional ScriptComponent script = 10;
}